src/output/scanline_output_file.rs

```diff
@@ -97,14 +52,14 @@ impl<'a, T: 'a + Write + Seek> ScanlineOutputFile<'a, T> {
                        owned: false,
                        _phantom: PhantomData,
                    },
-                   stream_writer: Some(ostream_ptr),
+                   ostream: ostream_ptr,
                    _phantom_1: PhantomData,
                    _phantom_2: PhantomData,
                })
         }
     }
 
-    pub fn write_pixels(&mut self, framebuffer: &FrameBuffer) -> Result<()> {
+    pub fn write_pixels(&mut self, framebuffer: &mut FrameBuffer) -> Result<()> {
```

Comment:
I believe `mut` is not needed here as during output the `FrameBuffer` is only read from, per previous discussion. Contrast the `InputFile::read_pixels` case, where `mut` is needed because the `FrameBuffer` is written to.